### ISSUE TITLE
optionally preserve hash order when loading [rt.cpan.org #83788]

### ISSUE THREAD

**Issue by ingydotnet** [ingydotnet/yaml-pm#132]
https://rt.cpan.org/Ticket/Display.html?id=83788

```
The current implementation of YAML does not allow to preserve the hash
order when loading a YAML document. I know, this is not really in the
spec, but:
- YAML documents come with an implicit order of the keys (as they appear
in the document)
- YAML::Node supports to keep the hash order via a tied hash

In fact, with the Loader changed on one line (around ~326):

-my $mapping = {};
+require YAML::Node;
+my $mapping = YAML::Node->new({});

... this works immediately! I am wondering whether this could not be
provided as a global option e.g. when $YAML::SortKeys is set to 0.

What do you think?

Many thanks,

Marek
```

**Comment by sergeyromanov** [comment]
It looks like this has been addressed by #9; tests are pending, though.

**Comment by perlpunk** [comment]
Released as `$YAML::Preserve` in 1.20
